### DISCUSSION TITLE
Display Events/Task from accounts related with Account Contact Relationship

### DISCUSSION THREAD

**Discussion by AnseGit** [deejay-hub/timeline-lwc#294] [Q&A] [answered]
Hi 
I need to display Events/Tasks related to PersonAccounts on Household Account Level (Financial Service Cloud). 
This does not appear to work with Open Activities and Activity History. 

Structure: 
We have a Household (Account-object), that has multiple Person Accounts related to it through Account Contact Relationship Junction Object. 
We need to display Events and Tasks related to all Person Accounts on Household-level.  

All these Events/Tasks are visible in Activity History - related list on Household - level, but they do not appear in Time warp. 
I have activated the metadata record for Open Activities and Activity History, but that did not solve this issue.

Do anyone know how to solve this?

**Comment by deejay-hub** [comment]
We can try and replicate if you could supply some screenshots with dummy records.

Here's what we tried.

1. Create 1 Household and 1 Person Account (Person One)
2. Associate them together
<img width="1155" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/d3cc2919-0f86-452c-85d6-f11b47c30a11">
3. Create a task for the Person Account (Person One)
<img width="1704" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/a1e71bd9-93d0-4ad6-ab0c-db7dd39d4c3c">
4. Navigate back to the Household to see if the Task is rolled up and see that the standard functionality doesn't do roll ups
<img width="1707" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/c681211d-6a57-42ed-a0c5-e4c334f48706">

So if we take Time Warp out of the picture for a second. Have we set things up correctly? If so the above gives no records returned in the standard timeline / activity history or open activities. You mentioned you saw the task in the activity history of the Household so we're curious what the steps are to replicate your scenario.

**Reply by AnseGit** [reply]
Hi 
Thx for quick response! 
You are replicating the issue correctly. 

I am able to view the record in the related list "Activity History" at household level, but not in the Activity timeline. 
Is it correct that we are not able to view task/event in Timewarp if they are not present in the Activity Timeline?
Is there another way of displaying tasks/event from all Person Account Activity Timeline on the Household Timewarp? 

![timewarp test](https://github.com/deejay-hub/timeline-lwc/assets/84069263/d5cd4249-4043-4199-baf4-6e784695d7cf)

**Reply by deejay-hub** [reply]
Can you explain how to get the Activities related list onto the page. We don't see the task in the ActivityHistory or OpenActivity standard objects.

Looks like you've added a 'Related List - Single' component to the page but I'm unsure of the values you use to populate the list

<img width="320" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/fb29d84b-4d79-441b-b257-e97d004d3bdd">

**Reply by AnseGit** [reply]
Correct as your picture above: 
![image](https://github.com/deejay-hub/timeline-lwc/assets/84069263/4bb15403-a363-4f1d-9ff3-e3c3c5004123)
![image](https://github.com/deejay-hub/timeline-lwc/assets/84069263/d81cd5e5-d213-41a9-b93b-f0b1f500f4fd)
![image](https://github.com/deejay-hub/timeline-lwc/assets/84069263/4227c0cd-fbba-4386-8588-d402bcf23796)

(We do have Financial Service Cloud in our Org. Not sure if that has something to do with why I get the record in the Activities-related list and not in the Activity Timeline)

**Comment by deejay-hub** [comment, marked as answer]
When you add a client to a Household there is an option to rollup Activities. When set to true the Activities related list is populated as you describe.

To display them in Time Warp you need to add the following record to the metadata.

<img width="1163" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/a2153a84-d5a4-407b-9515-15bf40425ab5">

You can deactivate OpenActivities and ActivityHistories.

**Reply by AnseGit** [reply]
This worked perfectly! Thank you!

**Reply by AnseGit** [reply]
For information:
After testing this solution with adding FinServ__Activities__r - custom metadata, we realised this solution did not work for users without System Admin Access. 

FinServ_Activities__r - Object is only queriable by users with admin access: 
![image](https://github.com/deejay-hub/timeline-lwc/assets/84069263/51b5421c-8edd-4e3b-bdbb-273d01d80436)
